test/unit/vl-select-location.test.html

```diff
@@ -272,6 +273,48 @@
         select.addEventListener('change', () => done());
         select.dispatchEvent(new CustomEvent('choice'));
       });
+
+      const getNoResultsText = (select) => {
+        return new Promise(async (resolve, reject) => {
+          await select.ready();
+          select._wrapperElement.addEventListener('click', async () => {
+            sendKey(select, 'P');
+            await awaitUntil(() => select._wrapperElement.querySelector('.has-no-results') != null);
+            resolve(select._wrapperElement.querySelector('.has-no-results').innerText);
+            sendKey(select, '');
+          });
+          select.focus();
+        });
+      };
+    
+      const getPlaceholderText = (select) => {
+        return new Promise(async (resolve, reject) => {
+          await select.ready();
+          return resolve(select._wrapperElement.querySelector('input').getAttribute('placeholder'));
+        });
+      };
+
+      const getNoMoreOptionsText = (select) => {
+        return new Promise(async (resolve, reject) => {
+          await select.ready();
+          select._wrapperElement.addEventListener('click', async () => {
+            sendKey(select, 'P');
+            sendBackspace(select);
+            await awaitUntil(() => select._wrapperElement.querySelector('.has-no-choices') != null);
+            resolve(select._wrapperElement.querySelector('.has-no-choices').innerText);
+          });
+          select.focus();
+        });
+      };
+
+      const sendBackspace = (select) => {
+        sendKey(select, '\b');
+      };
+    
+      const sendKey = (select, key) => {
+        select._wrapperElement.querySelector('input').value = key.charCodeAt(0) >= 32 ? key : '';
+        select._wrapperElement.querySelector('input').dispatchEvent(new KeyboardEvent('keyup', {'bubbles': true, 'keyCode': key.charCodeAt(0) || 46} ));
+      };
     });
```

Comment:
da stinkt hier naar duplicatie met de testen van vl-select ...
maar ik denk dat de enige manier is om dan ook een of ander test-util op unit test niveau ook te exposen in vl-select?

Reply:
ja, hier zit af en toe duplicatie met vl-select. Maar het is niet altijd 100% hetzelfde. GetNoMoreOptions  is hier nog anders.
Ik zal eens zien of ik de rest in een test util kan steken.

Reply:
Heb code weggedaan, omdat die in base klasse al wordt afgetest.